### ISSUE TITLE
Supports SAP Anywhere Database

### ISSUE THREAD

**Issue by sravangontla** [linq2db/linq2db#3398]
Hi Team,
 Can you please confirm that  linq2db supports SAP Anywhere Database. If yes please guide me, how to connect and so on...

note: 
   Inside SAP Anywhere parameterized query use ":" instead of "@"
   ex : ParameterName = ":Name";

More error details :
https://stackoverflow.com/questions/30688879/why-does-sql-anywhere-ignore-the-named-parameter

Looking for :
Asp.net core
Dapper
SAP SQL Anywhere (database)

**Comment by sravangontla** [comment]
Its not possible to use ?

**Comment by MaceWindu** [comment]
No, it requires database provider implementation. E.g. see existing providers
for ASE: https://github.com/linq2db/linq2db/tree/master/Source/LinqToDB/DataProvider/Sybase
or HANA: https://github.com/linq2db/linq2db/tree/master/Source/LinqToDB/DataProvider/SapHana

There is no active plans right now to work on new database providers in near future, so PRs are welcome.

**Comment by sravangontla** [comment]
Thanks for Quick Response.
Another Question: is it possible to inject custom provider from outside ?

**Comment by MaceWindu** [comment]
Yes, e.g. DB2 iSeries provider implemented as separate library https://github.com/LinqToDB4iSeries/Linq2DB4iSeries

**Comment by RoyChase** [comment]
I wrote the DB2 ISeries provider (based upon some POC code written by a another developer) and there a pros and cons to having it as separate library.  The initial effort for either doing it separatly or doing it as a part of Linq2Db (as a PR) are likely to be similar.  The benefit of doing it as a separate libary is that you control the release cadence.  If you find a bug you can fix it and release it whenever you wish.  Obviously if it is a part of Liqn2Db you are  dependant upon them.  

Where the external libary gives issues is when a new release of Linq2Db is published that you wish to use.  It's not uncommon for API's to change requiring changes in your provider.  So it's not always just a matter of pulling in the latest version of Linq2Db.  If you do it as a PR then it becomes a part of Linq2Db so it would just, I suspect, become another thing that gets refactored as a part of the normal developement process.
